libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -12,4 +13,19 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       _applicationId: applicationId,
     });
   }
+
+  async create(data: Partial<IntegrationEntity>): Promise<IntegrationEntity> {
+    const test = await this.find({
+      _applicationId: data._applicationId,
+      providerId: data.providerId,
+      channel: data.channel,
+    });
+    if (test.length === 0) {
```

Comment:
```suggestion
    if (existingIntegration) {
```

this can be modified to this after the above changes